go-controller/hybrid-overlay/pkg/windows/service/service.go

```diff
@@ -0,0 +1,111 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Based on https://github.com/kubernetes/kubernetes/blob/master/pkg/windows/service/service.go
+The reason for making a copy is that packages in kubernetes/kubernetes
+are not setup to be imported as modules without vendoring and pinning the sub
+components using the "require" directive. If more packages need to be imported
+then we can revisit importing this module too.
+*/
+
+package service
+
+import (
+	"os"
+	"time"
+
+	"k8s.io/klog"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+)
+
+var (
+	service *handler
+)
+
+type handler struct {
+	tosvc   chan bool
```

Comment:
Is this channel set somewhere? I am only seeing it being consumed but not set :)

Reply:
The channel is set in InitServie as follows: h := &handler{
		tosvc:   make(chan bool),
		fromsvc: make(chan error),
	}
you could be referring to something else here which I might be missing, please expand on it.

Reply:
I am seeing the channel is defined here, and as you mentioned initialized at line 51, and I am seeing the channel is read from at line 81, but I am not seeing any code that ever writes to this channel. Am I missing something?

Reply:
got it thanks

Reply:
done